hudi-common/src/main/java/org/apache/hudi/common/util/Option.java

```diff
@@ -123,6 +123,13 @@ public T orElse(T other) {
     return val != null ? val : other;
   }
 
+  /**
+   * Returns this {@link Option} if non-empty, and returns the given one otherwise
+   */
+  public Option<T> orElseOption(Option<T> other) {
+    return val != null ? this : other;
+  }
+
```

Comment:
Does this provide the same functionality as `public Option<T> or(Supplier<? extends Option<T>> other)` which can be reused?

Reply:
Oh, good catch!